src/model/world/world.cpp

```diff
@@ -4,28 +4,39 @@
 
 #include "world.h"
 
-World::World(int WIDTH, int HEIGHT, int ENERGY) : WIDTH{WIDTH}, HEIGHT{HEIGHT}, ENERGY{ENERGY} {
-    if (WIDTH <= 0 || HEIGHT <= 0 || ENERGY <= 0) throw std::range_error("World initialising values must be positive!");
-    fillUpEnergy();
+World::World(int WIDTH, int HEIGHT, int ENERGY) :
+    WIDTH{WIDTH},
+    HEIGHT{HEIGHT},
+    ENERGY{ENERGY}
+{
+    if (WIDTH <= 0) throw std::range_error("World(): WIDTH must be positive");
+    if (HEIGHT <= 0) throw std::range_error("World(): HEIGHT must be positive");
+    if (ENERGY <= 0) throw std::range_error("World(): ENERGY must be positive");
 }
 
-void World::addFood(int x, int y, int value) {
-    food.emplace_back(Food(x, y, value));
-    energy += value;
+World::~World() {
```

Comment:
for understanding purposes:
Is this the function being called when the object is destroyed?

Reply:
but thanks for reminding, I forgot deleting the individuals here